crates/shared/src/price_estimation/factory.rs

```diff
@@ -327,6 +332,7 @@ impl<'a> PriceEstimatorFactory<'a> {
         native: &[Vec<NativePriceEstimatorSource>],
         results_required: NonZeroUsize,
         weth: contracts::WETH9,
+        prices: Option<BTreeMap<H160, U256>>,
```

Comment:
Instead of requiring the prices for creating the estimator in the first place I think it would be nicer to have a method on the `CachingNativePriceEstimator` to initialize the cache after it got created.